source/postprocess/satellite_gravity.cc

```diff
@@ -0,0 +1,362 @@
+          depth_spacing = 0;
+          number_points_depth = 1;
+          number_points_longitude = (maximum_longitude - minimum_longitude) / mapping_spherical_spacing;
+          number_points_latitude = (maximum_latitude - minimum_latitude) / mapping_spherical_spacing;
+        }
+
+      // loop on r - satellite position [r, ,  ]
+      for (int h=0; h < number_points_depth; h++)
+        {
+          satellite_coordinate[0] = satellite_height + depth_spacing * h;
+          // CASE profile        -> satellite_height = 0 for first loop, then is incremental to profile_heigh_satellite
+          // CASE mapping        -> satellite_height = mapping_height_satellite
+          
+          // loop on phi - satellite position [ , phi ,]
+          for (int i=0; i < number_points_longitude; i++)
+            {
+              satellite_coordinate[1] = longitude + mapping_spherical_spacing * i;
+
+              // loop on theta - satllite position [ , , theta]
+              for (int j=0; j < number_points_latitude; j++)
+                { 
+                  satellite_coordinate[2] = latitude + mapping_spherical_spacing * j;
+                  
+                  // The spherical coordinates are shifted into cartesian to allow simplification in the mathematical equation.
+                  const Point<dim> position_satellite = Utilities::Coordinates::spherical_to_cartesian_coordinates<dim>(satellite_coordinate);
+                  cell = this->get_dof_handler().begin_active();
+                  double local_gx = 0;
+                  double local_gy = 0;
+                  double local_gz = 0;
+                  double gnorm = 0;
+                  double gtheory = 0;      
+                  double g_diff = 0;      
+                  double local_U = 0;
+                  number_cell_mpi = 0;
+              
+                  // For each point (i.e. satellite), the fourth integral goes over cells and quadrature 
+                  // points to get the unique distance between those, indispensable to calculate 
+                  // gravity vector components x,y,z, and potential. 
+                  for (; cell!=endc; ++cell)
+                    {
+                      if (cell->is_locally_owned())
+                        {
+                          fe_values.reinit (cell);
+                          const std::vector<Point<dim> > &position_point = fe_values.get_quadrature_points();
+
+                          for (unsigned int q = 0; q < number_quadrature_points_cell; ++q)
+                            {
+                              const double dist=(position_satellite[0]-position_point[q][0])*(position_satellite[0]-position_point[q][0]) +  
+                                                (position_satellite[1]-position_point[q][1])*(position_satellite[1]-position_point[q][1]) +  
+                                                (position_satellite[2]-position_point[q][2])*(position_satellite[2]-position_point[q][2]);  
```

Comment:
you can replace this by `(position_satellite - position_point).norm()`, easier to read :smile: